packages/rum-core/src/browser/windowOpenObservable.ts

```diff
@@ -0,0 +1,10 @@
+import { instrumentMethod, Observable } from '@datadog/browser-core'
+
+export function createWindowOpenObservable() {
+  return new Observable<void>((observable) => {
+    const { stop } = instrumentMethod(window, 'open', () => {
+      observable.notify()
+    })
+    return stop
+  })
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ It might fix the issue because we won't instrumentMethod more than one at a time, but it will instrumentMethod and stop intrumenting a lot (at every click)

I think your previous solution, with some improvement would be better:

```suggestion
let windowOpenObservable;

export function createWindowOpenObservable() {
  if (windowOpenObservable) {
    return windowOpenObservable;
  }
  
  const observable = new Observable<void>()
  const { stop: stopInstrumentation } = instrumentMethod(window, 'open', () => {
    observable.notify()
  })
  
  return { 
    observable, 
    stop() {
      stopInstrumentation()
      windowOpenObservable = undefined;
    }
  }
}
```